### PR TITLE
fix(docs): rework title in ComponentExample, fix layout

### LINKED ISSUE #977: Doc Site layout broken
In the Stardust Documentation the component example widths are seemingly random, off center, and being hidden behind the left nav.

image
image

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #977.

---

#### Before
image
#### After
image

### Reworked title

- much more compact controls
- more compact title
- plays well with Dark/HC theme

#### Before
image
#### After
image

### CODE REVIEW

File: docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx [hunk 1]
```diff
@@ -8,23 +8,15 @@ interface LabelledButtonProps {
...
+    const { iconName, label, onClick } = props
     return (
-      <div style={centered} onClick={onClick}>
-        <Icon
-          name={iconName}
-          styles={{ color: active ? 'green' : 'grey', marginBottom: '10px', border: 0 }}
-        />
-        <br />
-        <Text content={label} styles={{ color: active ? 'green' : 'grey' }} />
-      </div>
```
Comment (layershifter) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx:27]: Previously these styles were hardcoded and that give a strange result in Dark/HC themes.

### Before

image

### After

image

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 2]
```diff
@@ -40,90 +67,96 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
...
+                )}
+                value={anchorName}
+              />
+            ),
+            onClick: onCopyLink,
+            accessibility: toolbarButtonBehavior,
+          },
+        ]}
+      />
+    </Provider>
```
Comment (layershifter) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:159]: Only wrap with `Provider`

File: docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx [hunk 3]
```diff
@@ -71,41 +68,6 @@ class ComponentExample extends React.Component<ComponentExampleProps, ComponentE
...
-    match: PropTypes.object.isRequired,
-    title: PropTypes.node,
-    themeName: PropTypes.string,
-    exampleTheme: PropTypes.string,
-  }
-
-  componentWillMount() {
-    const { examplePath } = this.props
-    this.anchorName = examplePathToHash(examplePath)
-*/
```
Comment (layershifter) [inline docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx:108]: Cruft, looks like a wrong merge

File: docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx (hunk 1 above)
Comment (mnajdova) [reply docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx:27]: Can we fix the description of the example as well, to work well in all themes?

File: docs/src/components/ComponentDoc/ComponentExample/ComponentExampleTitle.tsx [hunk 4]
```diff
@@ -14,14 +14,10 @@ export default class ComponentExampleTitle extends React.PureComponent<Component
...
     return (
-      <div>
-        {title && (
-          <Header as="h3" className="no-anchor" styles={titleStyle}>
-            {title}
-          </Header>
-        )}
-        {description && <p>{description}</p>}
-      </div>
+      <Flex column>
```
Comment (mnajdova) [inline docs/src/components/ComponentDoc/ComponentExample/ComponentExampleTitle.tsx:17]: 👍

File: docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx (hunk 1 above)
Comment (layershifter) [reply docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx:27]: It comes from `Segment`'s styles as it's not defined in other themes.

Comment (layershifter) [reply docs/src/components/ComponentDoc/ComponentControls/ComponentButton.tsx:27]: image

https://github.com/stardust-ui/react/pull/1014/commits/f8581bd839bce3fd940c21c417e0dda9f00825ad 🎉

**Review by mnajdova — APPROVED** [review]

**Pull request merged**